src/Analyzers/MSTest.Analyzers/UseProperAssertMethodsAnalyzer.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+using Analyzer.Utilities.Extensions;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+using MSTest.Analyzers.Helpers;
+using MSTest.Analyzers.RoslynAnalyzerHelpers;
+
+namespace MSTest.Analyzers;
+
+/// <summary>
+/// MSTEST0037: Use proper 'Assert' methods.
+/// </summary>
+/// <remarks>
+/// The analyzer captures the following cases:
+/// <list type="bullet">
+/// <item>
+/// <code>Assert.[IsTrue|IsFalse](x [==|!=|is|is not] null)</code>
+/// </item>
+/// <item>
+/// <code>Assert.[IsTrue|IsFalse](x [==|!=] y)</code>
+/// </item>
+/// <item>
+/// <code>Assert.AreEqual([true|false], x)</code>
+/// </item>
+/// <item>
+/// <code>Assert.[AreEqual|AreNotEqual](null, x)</code>
+/// </item>
+/// </list>
+/// </remarks>
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+internal sealed class UseProperAssertMethodsAnalyzer : DiagnosticAnalyzer
+{
+    private enum NullCheckStatus
+    {
+        Unknown,
+        IsNull,
+        IsNotNull,
+    }
+
+    private enum EqualityCheckStatus
+    {
+        Unknown,
+        Equals,
+        NotEquals,
+    }
+
+    private static readonly LocalizableResourceString Title = new(nameof(Resources.UseProperAssertMethodsTitle), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString MessageFormat = new(nameof(Resources.UseProperAssertMethodsMessageFormat), Resources.ResourceManager, typeof(Resources));
+
+    internal static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.UseProperAssertMethodsRuleId,
+        Title,
+        MessageFormat,
+        null,
+        Category.Usage,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: false);
```

Comment:
@Evangelink I just remembered that we haven't discussed this. Is the current severity and being not enabled by default okay? Or should it be enabled by default?

Reply:
I think we can enable by default, I am only careful about the `warning` severity and outside of breaking change version I try to set it as `warning` only for rules where failure to comply would result in runtime error as we have many users with warn as error who would get broken by bumping.